lib/galaxy/datatypes/binary.py

```diff
@@ -3516,6 +3517,90 @@ def display_peek(self, dataset: DatasetProtocol) -> str:
             return f"PostgreSQL Archive ({nice_size(dataset.get_size())})"
 
 
+class MongoDBArchive(CompressedArchive):
+    """
+    Class describing a Mongo database packed into a tar archive
+
+    >>> from galaxy.datatypes.sniff import get_test_fname
+    >>> fname = get_test_fname('mongodb_fake.tar.bz2')
+    >>> MongoDBArchive().sniff(fname)
+    True
+    >>> fname = get_test_fname('test.fast5.tar')
+    >>> MongoDBArchive().sniff(fname)
+    False
+    """
+
+    MetadataElement(
+        name="version",
+        default=None,
+        param=MetadataParameter,
+        desc="MongoDB database version",
+        readonly=True,
+        visible=True,
+    )
+    file_ext = "mongodb"
+
+    def set_meta(self, dataset: DatasetProtocol, overwrite: bool = True, **kwd) -> None:
+        super().set_meta(dataset, overwrite=overwrite, **kwd)
+        try:
+            if dataset and tarfile.is_tarfile(dataset.file_name):
+                with tarfile.open(dataset.file_name, "r") as temptar:
+                    metrics_file = next(
+                        filter(lambda x: x.startswith("mongo_db/diagnostic.data/metrics"), temptar.getnames()), None
+                    )
+
+                    if metrics_file:
+                        md_version_file = temptar.extractfile(metrics_file)
+
+                        if md_version_file:
+                            vchunk = md_version_file.read(500)
+
+                            version_match = re.match(b".*version\x00\x06\x00\x00\x00([0-9.]+).*", vchunk)
+
+                            if version_match:
+                                dataset.metadata.version = util.unicodify(
+                                    version_match.group(1).decode("utf-8")
+                                ).strip()
+        except Exception as e:
+            log.warning("%s,(CompressedArchive set_meta Exception: %s", self, util.unicodify(e))
```

Comment:
```suggestion
            log.warning(f"{self} CompressedArchive set_meta Exception: {util.unicodify(e)}"
```
Did I get this right?